code/modules/mob/living/carbon/human/species_types/ethereal.dm

```diff
@@ -46,35 +55,118 @@
 	loreblurb = "Elzuosa are an uncommon and unusual species best described as crystalline, electrically-powered plantpeople. They hail from the warm planet Kalixcis, where they evolved alongside the Sarathi. Kalixcian culture places no importance on blood-bonds, and those from it tend to consider their family anyone they are sufficiently close to, and choose their own names."
 	var/drain_time = 0 //used to keep ethereals from spam draining power sources
 	var/obj/effect/dummy/lighting_obj/ethereal_light
+	var/datum/action/innate/root/rooting
 
 /datum/species/elzuose/Destroy(force)
 	if(ethereal_light)
 		QDEL_NULL(ethereal_light)
 	return ..()
 
-/datum/species/elzuose/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+/datum/species/elzuose/on_species_gain(mob/living/carbon/_carbon, datum/species/old_species, pref_load)
 	. = ..()
-	if(!ishuman(C))
+	if(!ishuman(_carbon))
 		return
-	var/mob/living/carbon/human/ethereal = C
+	var/mob/living/carbon/human/ethereal = _carbon
 	default_color = "#[ethereal.dna.features["ethcolor"]]"
 	RegisterSignal(ethereal, COMSIG_ATOM_EMAG_ACT, PROC_REF(on_emag_act))
 	RegisterSignal(ethereal, COMSIG_ATOM_EMP_ACT, PROC_REF(on_emp_act))
 	ethereal_light = ethereal.mob_light()
 	spec_updatehealth(ethereal)
+	rooting = new
+	rooting.Grant(_carbon)
+	RegisterSignal(ethereal, COMSIG_DIGOUT, PROC_REF(digout))
+	RegisterSignal(ethereal, COMSIG_MOVABLE_MOVED, PROC_REF(uproot))
 
 	//The following code is literally only to make admin-spawned ethereals not be black.
-	C.dna.features["mcolor"] = C.dna.features["ethcolor"] //Ethcolor and Mut color are both dogshit and will be replaced
-	for(var/obj/item/bodypart/BP as anything in C.bodyparts)
+	_carbon.dna.features["mcolor"] = _carbon.dna.features["ethcolor"] //Ethcolor and Mut color are both dogshit and will be replaced
+	for(var/obj/item/bodypart/BP as anything in _carbon.bodyparts)
 		if(BP.limb_id == SPECIES_ELZUOSE)
 			BP.update_limb(is_creating = TRUE)
 
-/datum/species/elzuose/on_species_loss(mob/living/carbon/human/C, datum/species/new_species, pref_load)
-	UnregisterSignal(C, COMSIG_ATOM_EMAG_ACT)
-	UnregisterSignal(C, COMSIG_ATOM_EMP_ACT)
+/datum/species/elzuose/on_species_loss(mob/living/carbon/human/_carbon, datum/species/new_species, pref_load)
+	UnregisterSignal(_carbon, COMSIG_ATOM_EMAG_ACT)
+	UnregisterSignal(_carbon, COMSIG_ATOM_EMP_ACT)
+	UnregisterSignal(_carbon, COMSIG_DIGOUT)
+	UnregisterSignal(_carbon, COMSIG_MOVABLE_MOVED)
 	QDEL_NULL(ethereal_light)
+	if(rooting)
+		rooting.Remove(_carbon)
 	return ..()
 
+/datum/action/innate/root
+	name = "Root"
+	desc = "Root into good soil to gain charge."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "plant-22"
+	icon_icon = 'icons/obj/flora/plants.dmi'
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/root/Activate()
```

Comment:
lowercase, `snake_case` is the standard versus `Pascal_Case` or `camel_Case`

Reply:
so turns out all the parent procs in action.dm are capitalized like this, and it's tied to a whole bunch of child procs. Would it be okay if I left this as is for now, and fix it in another PR, or should I change it here?